frontend/src/components/CommentForm.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from "react";
+import { Box, Button, useToast } from "@chakra-ui/react";
+import { Comment, postComment } from "../hooks/usePublicDecks";
+import { useUserContext } from "./UserContext";
+import InputFieldElement from "./InputFieldElement";
+
+interface CommentFormProps {
+  deckId: number;
+}
+
+const CommentForm = ({ deckId }: CommentFormProps) => {
+  const [commentText, setCommentText] = useState("");
+  const { token, username } = useUserContext();
+  const toast = useToast();
+
+  if (!token || !username) {
+    return null;
+  }
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!commentText.trim()) {
+      toast({
+        title: "Error",
+        description: "Comment cannot be empty.",
+        status: "error",
+        duration: 5000,
+        isClosable: true,
+      });
+      return;
+    }
+
+    const newComment: Comment = {
+      text: commentText,
+      username: username,
+    };
+
+    try {
+      const response = await postComment(deckId, newComment);
+      if (response.data) {
+        setCommentText("");
+
+        // Show success toast
+        toast({
+          title: "Comment Posted",
+          description: "Your comment has been posted successfully.",
+          status: "success",
+          isClosable: true,
+        });
+      } else if (response.error) {
+        toast({
+          title: "Error",
+          status: "error",
```

Comment:
En lidt mere sigende besked end "error" ville være fin her, for at indikere hvad der er galt